### PR TITLE
Switching views

### PR DESCRIPTION
**Pull request by Jayquintana**
Is this a fix or a feature?

-it is a feature 

What is the change?

-it add functionality to buttons, un hides and hides pages, switches between views of pages, pulls form for new 

What does it fix?

- nothing 

Where should the reviewer start?

- start at query selector variables

Were there any issues that arose?

- syntax errors,  UnHiding errors, hiding errors, button functionality.

GOALS?
 
- PR code goals is to add button  clicks, functionality, switch views from main page to create poster, and saved posters.

- future goals is to refactor this code, add the functions in a class? Clean up code in general

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -107,22 +128,68 @@ var savedPosters = [];
...
+
+
 // (we've provided one for you to get you started)!
 function getRandomIndex(array) {
+  // return Math.floor(Math.random() * array.length);
   return array[Math.floor(Math.random() * array.length)];
+ }
+
+
+ function randomPosterButton() {
```
Comment (ccollums) [inline src/main.js:153]: I think that we need to clean this up in the next iteration, possibly create a class here.

File: src/main.js [hunk 2]
```diff
@@ -107,22 +128,68 @@ var savedPosters = [];
...
+   imageInput.innerText = getRandomIndex(images);
+   imageInput.src = getRandomIndex(images);
+   titleInput.innerText = getRandomIndex(titles);
+   quotesInput.innerText = getRandomIndex(quotes);
+ }
+function mainPage() {
```
Comment (ccollums) [inline src/main.js:159]: same with this function, seems like we could make this a little cleaner with a class? or the already declared class file for poster?

File: src/main.js [hunk 3]
```diff
@@ -107,22 +128,68 @@ var savedPosters = [];
...
+ imageInput.src = getRandomIndex(images);
+ titleInput.innerText = getRandomIndex(titles);
+ quotesInput.innerText = getRandomIndex(quotes);
 }
-function randomPosterButton() {
-  imageInput.innerText = getRandomIndex(images);
-  imageInput.src = getRandomIndex(images);
-  titleInput.innerText = getRandomIndex(titles);
-  quotesInput.innerText = getRandomIndex(quotes);
+mainPage();
```
Comment (ccollums) [inline src/main.js:165]: does this mainPage() need to be invoked here?

File: src/main.js [hunk 4]
```diff
@@ -1,9 +1,30 @@
 // query selector variables go here 👇
-
+// Image, Title, Quotes
 var imageInput = document.querySelector('.poster-img');
 var titleInput = document.querySelector('.poster-title');
 var quotesInput = document.querySelector('.poster-quote');
-var showRandomPoster = document.querySelector('.show-random');
+
+// show random, make poster buttons, saved poster Button
```
Comment (ccollums) [inline src/main.js:7]: in our final push we will need to remove all these commented out lines, but I like it now while we work on the project!

File: src/main.js [hunk 5]
```diff
@@ -107,22 +128,68 @@ var savedPosters = [];
...
-showRandomPoster.addEventListener('click', randomPosterButton);
+showRandomButton.addEventListener('click', randomPosterButton);
+// make a poster button
+makePosterButton.addEventListener('click',hideMainShowForm);
+// saved posters sections
+savedPostersButton.addEventListener('click', showSaved);
+// go back to main buttons
+nevermindButton.addEventListener('click', backToMainPage);
+backToMainButton.addEventListener('click', backToMainPage);
+
```
Comment (itsnameissteven) [inline src/main.js:139]: These look good. Nice use of named functions rather than anonymous functions

File: src/main.js (hunk 2 above)
Comment (itsnameissteven) [inline src/main.js:159]: This should be an action verb 'do a thing' . I think here maybe createRandomImage or similar makes more sense

File: src/main.js [hunk 6]
```diff
@@ -107,22 +128,68 @@ var savedPosters = [];
...
 }
 
-imageInput.innerText = getRandomIndex(images);
-imageInput.src = getRandomIndex(images);
-titleInput.innerText = getRandomIndex(titles);
-quotesInput.innerText = getRandomIndex(quotes);
+function showSavedPosters() {
+  savedPostersPage.classList.remove('hidden');
+}
+
```
Comment (itsnameissteven) [inline src/main.js:179]: These are good, but if you wanted to you could write a reusable function. When I write out functions like this and see they are very similar I like to refactor them into one. There is a parameter (or two) you can use to make a very dynamic function. You could even use that function inside other functions

**Review by itsnameissteven — COMMENTED** [review]
Overall I think the code is pretty clean. I don't see any functionality issues. I like all the code comments too. 

I pointed out a few spots that could use a refactor if you would like, not 100% necessary but would clean things up a little bit.

**Pull request merged**